### PR TITLE
[MRG] Use `fasterq-dump` instead of `fastq-dump` to enable multithreaded downloads

### PR DESCRIPTION
**Pull request by bluegenes**
[`fasterq-dump`](https://github.com/ncbi/sra-tools/wiki/HowTo:-fasterq-dump) has a few benefits over `fastq-dump`
- multithreaded downloads (6 is default, I enabled up to 8 in this PR)
- enables setting temp dir without changing directories
- renaming to sample name happens automatically
- better defaults:
  - `split-3` enabled by default. I found `--split-files` to work better in a few cases, used here
  - `skip-technical` enabled by default

TBD::
- Do we need these options?
  - `split-spot`? 
    - It can be used with `fasterq-dump` but in my brief test, 
      it produced `_1.pass.fastq` and `.fastq` files instead of `_1.fastq` and `_2.fastq` (which were produced with default options)
  - `clip`-- not an option
  -  `dumpbase` --not an option
  - `--readids` -- not an option

I think it's fine to exclude these. There's no mention of using them in the [docs](https://github.com/ncbi/sra-tools/wiki/08.-prefetch-and-fasterq-dump).

- Single end downloads should go to `*.fastq` file, and no `_1` /`_2` files. Touch unpaired file if paired files are downloaded. If not and an unpaired file has been downloaded, touch paired files.

### CODE REVIEW

**Comment by bluegenes** [comment]
I think this is ready for review @ctb @taylorreiter.

File: genome_grist/conf/Snakefile [hunk 1]
```diff
@@ -208,79 +208,74 @@ rule zip:
...
 rule download_sra_wc:
     output:
-        r1 = protected(outdir + "/raw/{sample}_1.fastq.gz"),
-        r2 = protected(outdir + "/raw/{sample}_2.fastq.gz"),
+        r1  = protected(outdir + "/raw/{sample}_1.fastq.gz"),
+        r2  = protected(outdir + "/raw/{sample}_2.fastq.gz"),
         unp = protected(outdir + "/raw/{sample}_unpaired.fastq.gz"),
         temp_dir = temp(directory(f"{base_tempdir}/{{sample}}.d")),
+    params:
+        temp_r1 = lambda w: f"{base_tempdir}/{w.sample}.d/{w.sample}_1.fastq",
```
Comment (ctb) [inline genome_grist/conf/Snakefile:216]: why do you need to use lambda here? can't you use wildcards directly? I guess this works but seems overly complicated.

I'm thinking something like:

```
temp_r1 = f"{base_tempdir}/{{wildcards.sample}}.d/{{wildcards.sample}}_1.fastq"
```

**Review by ctb — APPROVED** [review]
looks great to me! I'll be interested in Taylor's review, too.

File: genome_grist/conf/Snakefile (hunk 1 above)
Comment (bluegenes) [reply genome_grist/conf/Snakefile:216]: You can't access wildcards in params without lambda, right? But we could make them temporary outputs

Comment (taylorreiter) [reply genome_grist/conf/Snakefile]: I'm not sure if this had been updated in recent versions of snakemake but yes, historically you could not use wildcards in params without lambda bc of the order of execution of the DAG

**Review by taylorreiter — COMMENTED** [review]
i know this is already merged but I agree, looks great. nice to have fasterq dump rules laid out in a workflow. I've always used ftp fastq download from ENA link in workflows bc i hate dealing with fastq dump, but this seems more reliable than that.

**Comment by bluegenes** [comment]
@taylorreiter the only thing I'd say is that every now and then, the `_1` and `_2` don't seem to split properly. As we talked about, some of the samples seem to be incorrect on the SRA. I'm not 100% sure that's true in every case though -- something to keep an eye on as we go forward using this!

**Comment by bluegenes** [comment]
also, a somewhat annoying issue with writing `protected` files here, is that if the workflow does not exit properly and there are un-deleted, unfinished, protected files, they need to be manually deleted for the workflow to proceed. I'm hoping this was a one-off for some cancellations I got impatient with, but if you experience this too, we can make an issue and workaround.

**Pull request merged**